task-6-express/src/controllers/cart.controller.ts

```diff
@@ -0,0 +1,45 @@
+import { Request, Response } from 'express';
+import CartService from '../services/cart.service';
+
+const cartService = new CartService();
+
+export const getUserCart = async (req: Request, res: Response) => {
+  try {
+    const userId = req.headers['x-user-id'] as string;
+    const cart = await cartService.getUserCart(userId);
+    res.status(200).json({ data: cart, error: null });
+  } catch (error) {
+    res.status(500).json({ data: null, error: { message: 'Internal Server Error' } });
+  }
+};
+
+export const updateUserCart = async (req: Request, res: Response) => {
+  try {
+    const userId = req.headers['x-user-id'] as string;
+    const cartSnapshot = req.body;
+    const updatedCart = await cartService.updateUserCart(userId, cartSnapshot);
+    res.status(200).json({ data: updatedCart, error: null });
```

Comment:
We can create a separate object for statuses and use them instead of hardcoded number:

```
const RESPONSE_STATUS = {
  SUCCESS: 200,
  ERROR: 500,
  NOT_FOUND: 404,
  NOT_AUTHORIZED: 401,
  ...etc
};
```